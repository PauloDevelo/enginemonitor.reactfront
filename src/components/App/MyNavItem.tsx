
import React from 'react';
import { NavItem, NavLink } from 'reactstrap';
import { FormattedMessage,  } from 'react-intl';

import { useTraceUpdate } from '../../hooks/Debug';

type Props = {
    classNames: string,
    activeFunc: () => void, 
    label: FormattedMessage.MessageDescriptor
};

const MyNavItem = ({classNames, activeFunc, label}:Props) => {
    useTraceUpdate({classNames, activeFunc, label});

    return <NavItem>
                <NavLink className={classNames} 
                    onClick={activeFunc}>
                    <FormattedMessage {...label}/>
                </NavLink>
            </NavItem>
} 

export default React.memo(MyNavItem);