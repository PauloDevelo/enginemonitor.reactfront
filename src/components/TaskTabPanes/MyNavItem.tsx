import React from 'react';
import { NavItem, NavLink } from 'reactstrap';
// eslint-disable-next-line no-unused-vars
import { FormattedMessage, MessageDescriptor } from 'react-intl';

type Props = {
    classNames: string,
    activeFunc: () => void,
    label: MessageDescriptor
};

const MyNavItem = ({ classNames, activeFunc, label }:Props) => (
  <NavItem>
    <NavLink
      className={classNames}
      onClick={activeFunc}
    >
      <FormattedMessage {...label} />
    </NavLink>
  </NavItem>
);

export default React.memo(MyNavItem);
