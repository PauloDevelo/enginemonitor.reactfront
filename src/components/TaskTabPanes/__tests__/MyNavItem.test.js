import React from 'react';
import { mount } from 'enzyme';

import ignoredMessages from '../../../testHelpers/MockConsole';

import MyNavItem from '../MyNavItem';

describe('MyNavItem', () => {
  beforeAll(() => {
    ignoredMessages.length = 0;
    ignoredMessages.push('[React Intl] Could not find required `intl` object.');
  });

  beforeEach(() => {
  });

  it('should render a MyNavItem', () => {
    // Arrange
    const message = { id: 'tooltip', defaultMessage: 'hello world', description: 'a tooltip' };

    // Act
    const wrapper = mount(<MyNavItem label={message} />);

    // Assert
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the function onClick when the user click on the tab', () => {
    // Arrange
    const message = { id: 'tooltip', defaultMessage: 'hello world', description: 'a tooltip' };
    const activeFunc = jest.fn();

    const wrapper = mount(<MyNavItem label={message} activeFunc={activeFunc} />);

    // Act
    wrapper.find('NavLink').simulate('click');

    // Assert
    expect(activeFunc).toBeCalledTimes(1);
  });
});
