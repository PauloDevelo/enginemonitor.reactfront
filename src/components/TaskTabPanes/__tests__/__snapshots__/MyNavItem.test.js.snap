// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyNavItem should render a MyNavItem 1`] = `
<Memo(MyNavItem)
  label={
    Object {
      "defaultMessage": "hello world",
      "description": "a tooltip",
      "id": "tooltip",
    }
  }
>
  <NavItem
    tag="li"
  >
    <li
      className="nav-item"
    >
      <NavLink
        tag="a"
      >
        <a
          className="nav-link"
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="hello world"
            description="a tooltip"
            id="tooltip"
            values={Object {}}
          >
            hello world
          </FormattedMessage>
        </a>
      </NavLink>
    </li>
  </NavItem>
</Memo(MyNavItem)>
`;
